src/Solvers/conjugate_gradient_poisson_solver.jl

```diff
@@ -1,4 +1,5 @@
 using Oceananigans.Operators: divᶜᶜᶜ, ∇²ᶜᶜᶜ 
+using Oceananigans.Grids: ImmersedBoundaryGrid
```

Comment:
Hmm, I think `ImmersedBoundaryGrid` is defined by `ImmersedBoundaries`, which isn't imported until after `Solvers`:

https://github.com/CliMA/Oceananigans.jl/blob/99d3c1fcb8cb7951c0b82f82d9d61d2fe9d541e1/src/Oceananigans.jl#L210-L227

A bit of refactoring is needed to import `ImmersedBoundaries` before `Solvers`. So I think if we want to resolve this without all that refactoring we should define the constructor that depends on `ImmersedBoundaryGrid` separately (either in `ImmersedBoundaries` or `Model.NonhydrostaticModels`)

Reply:
yeah I think you're right!